bin/crawler.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+#[macro_use]
+extern crate tracing;
+
+use snarkos::{
+    cli::CLI,
+    config::{Config, ConfigCli},
+    display::{initialize_logger, print_welcome},
+    errors::NodeError,
+};
+use snarkos_network::{config::Config as NodeConfig, Node};
+use snarkos_rpc::start_rpc_server;
+
+use std::{net::SocketAddr, time::Duration};
+
+use tokio::runtime;
+
+///
+/// Builds a node from configuration parameters.
+///
+/// 1. Creates network server.
+/// 2. Starts rpc server thread.
+/// 3. Starts network server listener.
+///
+async fn start_server(config: Config) -> anyhow::Result<()> {
+    initialize_logger(&config);
+
+    print_welcome(&config);
+
+    let address = format!("{}:{}", config.node.ip, config.node.port);
+    let desired_address = address.parse::<SocketAddr>()?;
+
+    let mut path = config.node.dir.clone();
+    path.push(&config.node.db);
+
+    let node_config = NodeConfig::new(
+        desired_address,
+        config.p2p.min_peers,
+        config.p2p.max_peers,
+        config.p2p.bootnodes.clone(),
+        config.node.is_bootnode,
+        config.node.is_crawler,
```

Comment:
```suggestion
        false,
        true,
```

Reply:
I'd keep this here, since we're setting the node type on `Config` in `main` anyway—so long as configuration is coupled at least?